docs/source/python/getstarted.rst

```diff
@@ -0,0 +1,149 @@
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _getstarted:
+
+Getting Started
+===============
+
+Arrow manages data in Arrays (:class:`pyarrow.Array`), which can be
+grouped in tables (:class:`pyarrow.Table`) to represent columns of data
+in tabular data.
+
+Arrow also exposes supports for various formats to get those tabular
+data in and out of disk and networks. Most commonly used formats are
+Parquet (:ref:`parquet`) and the IPC format (:ref:`ipc`). 
+
+Creating Arrays and Tables
+--------------------------
+
+Arrays in Arrow are collections of data of uniform type. That allows
+arrow to use the best performing implementation to store the data and
+perform computation of it. So each array is meant to have data and
+a type
+
+.. ipython:: python
+
+    import pyarrow as pa
+
+    days = pa.array([1, 12, 17, 23, 28], type=pa.int8())
+
+multiple arrays can be combined in tables to form the columns
+in tabular data according to a provided schema
+
+.. ipython:: python
+
+    months = pa.array([1, 3, 5, 7, 1], type=pa.int8())
+    years = pa.array([1990, 2000, 1995, 2000, 1995], type=pa.int16())
+
+    birthdays_table = pa.table([days, months, years], 
+                               schema=pa.schema([
+                                    ('days', days.type),
+                                    ('months', months.type),
+                                    ('years', years.type)
+                               ]))
+    
+    birthdays_table
+
+See :ref:`data` for more details.
+
+Saving and Loading Tables
+-------------------------
+
+Once you have a tabular data, Arrow provides out of the box
+the features to save and restore that data for common formats
+like parquet
+
+.. ipython:: python   
+
+    import pyarrow.parquet as pq
+
+    pq.write_table(birthdays_table, 'birthdays.parquet')
+
+Once you have your data on disk, loading it back is as easy,
+and Arrow is heavily optimized for memory and speed so loading
+data will be as quick as possible
+
+.. ipython:: python
+
+    reloaded_birthdays = pq.read_table('birthdays.parquet')
+
+    reloaded_birthdays
+
+Saving and loading back data in arrow is usually done through
+:ref:`parquet`, :ref:`ipc`, :ref:`csv` or :ref:`json` formats.
+
+Performing Computations
+-----------------------
+
+Arrow ships with a bunch of compute functions that can be applied
+to its arrays, so through the compute functions it's possible to apply
+transformations to the data
+
+.. ipython:: python
+
+    import pyarrow.compute as pc
+
+    pc.value_counts(birthdays_table["years"])
+
+See :ref:`compute` for a list of available compute functions and
+how to use them.
+
+Working with big data
```

Comment:
I would personally avoid the term "Big Data" (although you maybe didn't mean "Big Data" with capitals here :), it's still what many people will read)

Not directly sure what would be a good alternative. "Working with datasets" is probably not specific enough. "Working with partitioned datasets"? (or "Working with larger, partitioned datasets" ..)

Reply:
I see what you mean, but I explicitly avoided using "partitioned data" because I perceive partitioning as the solution to a problem, not the problem itself. And here I was looking for a title that described what the following paragraphs are going to explain how to solve.

I guess we can use "large data"? instead of "big data"? That doesn't change much from the title meaning point of view but it would avoid using a word that people explicitly map to something specific.

Reply:
(personally I would be in favor of using "large" instead of "big", that seems a bit more neutral)

Reply:
:+1: renamed "big" to "large"